src/Explorer/Tabs/CloudShellTab/README.md

```diff
@@ -0,0 +1,282 @@
+   - Checks if CloudShell is enabled via `userContext.features.enableCloudShell`
+   - Generates appropriate terminal buttons based on database type
+   - Handles conditional rendering of CloudShell options
+
+8. **TerminalTab**: Container component for terminal experiences
+   - Renders appropriate terminal type based on the selected database
+   - Manages terminal tab state and lifecycle
+   - Provides the integration point between the terminal and the rest of the Cosmos Explorer UI
+
+## Telemetry Collection
+
+CloudShell components utilize `TelemetryProcessor.trace` to collect usage data and diagnostics information that help improve the service and troubleshoot issues.
+
+### Telemetry Events
+   - When CloudShell Starts
+   - User Consent to access  shell out of the region
+   - When shell is connected
+   - When there is an error during CloudShell initialization
+
+| Action Name | Description | Collected Data |
+|------------|------------|----------------|
+| CloudShellTerminalSession/Start | Triggered when user starts a CloudShell session | Shell Type, dataExplorerArea as <i>CloudShell</i>|
+| CloudShellUserConsent/(Success/Failure) | Records user consent to get cloudshell in other region |  |
+| CloudShellTerminalSession/Success | Records if Terminal creation is successful | Shell Type, Shell Region |
+| CloudShellTerminalSession/Failure | Records of terminal creation is failed | Shell Type, Shell region (if available), error message  |
+
+### Real-time Use Cases
+
+1. **Performance Monitoring**:
+   - Track shell initialization times across different regions and database types
+
+2. **Error Detection and Resolution**:
+   - Detect increased error rates in real-time
+   - Identify patterns in failures
+   - Correlate errors with specific client configurations
+
+3. **Feature Adoption Analysis**:
+   - Measure adoption rates of different terminal types
+
+4. **User Experience Optimization**:
+   - Analyze session duration to understand engagement
+   - Identify abandoned sessions and potential pain points
+   - Measure the impact of new features on usage patterns
+   - Track command completion rates and error recovery
+
+## Limitations and Regional Availability
+
+### Network Isolation
+
+Network isolation (such as private endpoints, service endpoints, and VNet integration) is not currently supported for CloudShell connections. All connections to database instances through CloudShell require the database to be accessible through public endpoints.
+
+Key limitations:
+- Cannot connect to databases with public network access disabled
+- No support for private link resources
+- No integration with Azure Virtual Networks
+- IP-based firewall rules must include CloudShell service IPs
+
+### Data Residency
+
+Data residency requirements may not be fully satisfied when using CloudShell due to limited regional availability. CloudShell services are currently available in the following regions:
```

Comment:
How do we check for updated Cloudshell availability ?

Reply:
Right now, we are relying on cloudshell team for this information, may be in sometime they will provide an API for this information. I will give them this feedback.